Gcp flight recorder

kapilt opened the pull request.

thisisshi commented:
Wow the `get-discovery-v1-apis-compute-v1-rest_1.json` recorded json is 31k lines... should the recorder minify/compress these recorded json files for the sake of the size of the repo or look into using submodules or something?

kapilt commented:
good idea.. yeah we could probably gzip on the discovery files to minimize some of the record delta.

kapilt commented:
switched out discovery files to bzip2, which shrinks the on disk from 1.6mb to 60k, and should remove them from the pr line diff stats. bz2 decompress was a bit expensive on test speed, so also added in an in mem cache on the discovery files.
